### ISSUE TITLE
Multiselect onChange property not working

### ISSUE THREAD

**Issue by noire-munich** [mantinedev/mantine#1857]
### What package has an issue

@mantine/core

### Describe the bug

When using `MultiSelect`, the `onChange` callback is not triggered.
Oddly enough, an `onSelect` is triggered.

I checked the source code, which confirms the docs in saying that the property to use should be `onChange`.

This is happening while trying to use Mantine with `react-hook-forms` within a RedwoodJS project.

Code to check:
```
import { MultiSelect } from "@mantine/core";
import "./styles.css";

export default function App() {
  return (
    <div className="App">
      <MultiSelect
        label="This uses onChange and triggers nothing."
        onChange={(x) => {
          console.log(x);
        }}
        data={["red", "white", "blue"]}
      />
      <MultiSelect
        label="This uses onSelect and output to console."
        onSelect={(x) => {
          console.log(x);
        }}
        data={["red", "white", "blue"]}
      />
    </div>
  );
}
```

### In which browser did the problem occur

Chrome/Linux

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/reverent-snowflake-sv9iqx?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

**Comment by rtivital** [comment]
From version 5.0 it is required to provide both `value` and `onChange` for controlled components (for all components, not only MultiSelect). If you do not provide value then onChange function will not be called.

https://codesandbox.io/s/intelligent-kare-rib3yn?file=/src/App.tsx

**Comment by noire-munich** [comment]
Thanks @rtivital , indeed it works better now.

Maybe this should be made more obvious in the doc, otherwise there's nothing really telling you that `value` is required when `onChange` is used.
The hook https://github.com/mantinedev/mantine/blob/c7d080c2133b0196e3a8382ec6134838632c8f9a/src/mantine-hooks/src/use-uncontrolled/use-uncontrolled.ts#L17 shows it, but the doc doesn't.

**Comment by pstachula-dev** [comment]
@rtivital Have the same issue:

`target.value` is equal 'blue' instead of 2

```
 <Select
        label="This uses onSelect and output to console."
        onSelect={(e) => {
          console.log(e.target.value);
        }}
        data={[
          {
            value: 2,
            label: "blue"
          }
        ]}
      />
```

**Comment by rtivital** [comment]
Select component does not support `onSelect` prop.

**Comment by steschwa** [comment]
I agree that this behaviour doesn't feel "right". I think the main problem resides in [`use-uncontrolled.ts`](https://github.com/mantinedev/mantine/blob/master/src/mantine-hooks/src/use-uncontrolled/use-uncontrolled.ts#L23). If the `value` prob of `<Select>` is `undefined` at the first render, the component keeps acting like it's uncontrolled because it'll never switch back to beeing controlled.

If we compare that to e.g. [useControllableState.ts](https://github.com/radix-ui/primitives/blob/main/packages/react/use-controllable-state/src/useControllableState.tsx) from Radix, they compute the controlled/uncontrolled state on the fly.

**Comment by xorander00** [comment]
Agreed. I just ran into this issue too. In my case, on first render I'd like the select to not have a default value (undefined). I also want the control to be clearable, so my only options are to either add an empty item or pre-select one of the existing items (though in my specific case that's likely to cause a UX issues).

**Comment by rtivital** [comment]
Fixed in 5.0.2, `onChange` is now called for uncontrolled component as well
